homeassistant/components/iaqualink/__init__.py

```diff
@@ -215,18 +215,23 @@ class AqualinkEntity(Entity):
     def __init__(self, dev: AqualinkDevice) -> None:
         """Initialize the entity."""
         self.dev = dev
+        self._attr_unique_id = f"{self.dev.system.serial}_{self.dev.name}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self._attr_unique_id)},
+            manufacturer=dev.manufacturer,
+            model=dev.model,
+            # Instead of setting the device name to the entity name, iaqualink
+            # should be updated to set has_entity_name = True
+            name=cast(str | None, self.name),
```

Comment:
One downside is, the name isn't set at this point yet. Later in your PR, `_attr_name` is set after the super constructor, so when this is executed name is undefined

Reply:
I ran the tests in debug mode. And from there it seems same is set.

Reply:
Change it. May be because this because we use super now.